docs/projected-volume.md

```diff
@@ -0,0 +1,27 @@
+# Projected Volumes and Windows Pods
+
+A [projected volume](https://kubernetes.io/docs/concepts/storage/volumes/#projected) allows mapping of several existing
+volume sources into the same directory in the container. With the [proposal for file permission handling in projected
+service account volume](https://github.com/kubernetes/enhancements/pull/1598) enhancement, the projected files have
+the correct permissions set including container user ownership for Linux pods when `RunAsUser` is set in the Pod 
+`SecurityContext`. However, for Windows pods this is not the case when `RunAsUsername` is set, due to differences in 
+the way user accounts are managed in Windows. Windows stores and manages local user and group accounts in a database
+file called Security Account Manager (SAM). This database is not shared between the Windows container host and the
+running containers. This prevents the kubelet from setting correct ownership on the files in the projected volume.
+This problem is tracked by [Windows Pod with RunAsUserName and a Projected Volume does not honor file
+permissions in the volume](https://github.com/kubernetes/kubernetes/issues/102849) issue.
+
+##Bound Service Account Token Volume
+The [Bound Service Account Token Volume](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume)
+feature results in the ServiceAccount admission controller adding a projected volume instead of a Secret-based volume
+for the non-expiring service account token created by the Token Controller. This implies that on all Kubernetes 1.22+ 
+clusters, a projected volume with service account token will be present on all Pods. Now for Windows Pods`RunAsUserName`
+is set in the Pod `SecurityContext`, the ownership will not be set correctly. This problem can get exacerbated when 
+used in conjunction with [HostPath](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath) where best practices
+are not followed. For example, giving a Pod access to ` c:\var\lib\kubelet\pods\` will result in that Pod being able to
+access service account tokens from other Pods.
+
+##RunAsUser and RunAsUsername
+Creating a Windows Pod with RunAsUser and RunAsUsername in it's `SecurityContext` will result in the Pod being stuck at
```

Comment:
This is the state of windows pods on openshift in 4.8?  Is there a benefit to allowing such pods through admission or should we ship a patch on 4.8 or 1.21 to directly fail when a windows pod has this combination of thing?

Reply:
In OpenShift 4.8 we introduced a workaround https://github.com/openshift/kubernetes/pull/804 for the combination.

Reply:
> In OpenShift 4.8 we introduced a workaround [openshift/kubernetes#804](https://github.com/openshift/kubernetes/pull/804) for the combination.

This should be explained here, along with the impact.  If I've understood it all correctly, the workaround created in 804 means that RunAsUser and RunAsUsername are not honored for projected volumes on Windows.  IOW, it fails open.  I think that we chose to fail open rather than to perma-fail pods in the manner you describe on these lines.

If so, can we link the PR, describe the behavior and explain the alternative we avoided?

Reply:
>If I've understood it all correctly, the workaround created in 804 means that RunAsUser and RunAsUsername are not honored for projected volumes on Windows.

To be clear even if just `RunAsUsername` is specified, the situation is the same i.e. the ownership is not set to username specified. However because the default permission for regular users is `BUILTIN\Users Allow  ReadAndExecute, Synchronize`, the `RunAsUsername` will have read access to files.

And yes, I will call this out and link to the PR.

Reply:
I have now updated the PR with this info,